File: Test/comp/ExternCtors-externs/Library.go
// FIXME: Test currently fails on Go

package Library

import (
	"dafny"
	"fmt"
)

type Class struct{ n dafny.Int }

func New_Class_(n dafny.Int) *Class {
	return &Class{n}
}

func (obj *Class) Get() dafny.Int {
	return obj.n
}

// have to implement this because the Go backend can't mix Dafny and Go code :-\

func (obj *Class) Print() {
	fmt.Printf("My value is %d\n", obj.n)
}

type CompanionStruct_Class_ struct{}
var Companion_Class_ = CompanionStruct_Class_{}

func (CompanionStruct_Class_) SayHi() {
	fmt.Println("Hello!");
}

File: Test/comp/ExternCtors-externs/Library.py
import sys

import module_
import _dafny
import System_

assert "Library" == __name__
Library = sys.modules[__name__]

class Class:
    def __init__(self, n) -> None:
        self.n = n

    @staticmethod
    def SayHi():
        print("Hello!")

    def Get(self):
        return self.n

    def Print(self):
        _dafny.print(_dafny.string_of(_dafny.Seq("My value is ")))
        _dafny.print(_dafny.string_of((self).Get()))
        _dafny.print(_dafny.string_of(_dafny.Seq("\n")))
